File: src/Calculator.Shared/Common/AppResourceProvider.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using Windows.ApplicationModel.Resources;

namespace CalculatorApp
{
    public class AppResourceProvider
    {
        Windows.ApplicationModel.Resources.ResourceLoader m_stringResLoader;
        Windows.ApplicationModel.Resources.ResourceLoader m_cEnginestringResLoader;
        private static AppResourceProvider s_instance;

        public AppResourceProvider()
        {
            m_stringResLoader = ResourceLoader.GetForViewIndependentUse();
            m_cEnginestringResLoader = ResourceLoader.GetForViewIndependentUse("CEnginestrings");
        }

        public static AppResourceProvider GetInstance()
        {
            if(s_instance == null)
            {
                s_instance = new AppResourceProvider();
            }

            return s_instance;
        }

        public string GetResourceString(string key)
        {
            return m_stringResLoader.GetString(key);
        }

        public string GetCEngineString(string key)
        {
            return m_cEnginestringResLoader.GetString(key);
        }
    }
}

File: src/CalcViewModel/Common/AppResourceProvider.cpp
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

#include "pch.h"
#include "AppResourceProvider.h"

using namespace Platform;
using namespace Windows::ApplicationModel::Resources;
using namespace CalculatorApp;
using namespace CalculatorApp::ViewModel::Common;

AppResourceProvider::AppResourceProvider()
{
    m_stringResLoader = ResourceLoader::GetForViewIndependentUse();
    m_cEngineStringResLoader = ResourceLoader::GetForViewIndependentUse(L"CEngineStrings");
}

AppResourceProvider ^ AppResourceProvider::GetInstance()
{
    static AppResourceProvider ^ s_instance = ref new AppResourceProvider();
    return s_instance;
}

String ^ AppResourceProvider::GetResourceString(_In_ String ^ key)
{
    return m_stringResLoader->GetString(key);
}

String ^ AppResourceProvider::GetCEngineString(_In_ String ^ key)
{
    return m_cEngineStringResLoader->GetString(key);
}
